server/jetstream_api.go

```diff
@@ -922,8 +943,10 @@ func (s *Server) processJSAPIRoutedRequests() {
 	for {
 		select {
 		case <-queue.ch:
-			reqs := queue.pop()
-			for _, r := range reqs {
+			// Only pop one item at a time here, otherwise if the system is recovering
+			// from queue buildup, then one worker will pull off all the tasks and the
+			// others will be starved of work.
+			for r, ok := queue.popOneLast(); ok && r != nil; r, ok = queue.popOneLast() {
```

Comment:
I was going to say that this seems wrong: picture a `push("some create operation")` followed by `push("some delete of that created object")`. With `popOneLast()`, you would possibly get the "delete" first, then the "create". But I realize that a change was made some time ago to make `processJSAPIRoutedRequests()` be executed from several go routines, which then already completely put the "ordering" out of the window. So I assume that if that was done it means that those actions are independent of each other and order does not matter. You would want to verify that this is the case. Again, if there is possibly ordering issue, then `popOneLast()` would be wrong, but running `processJSAPIRoutedRequests()` from more than 1 go routine would be wrong too.

Reply:
Indeed, the ordering is already indeterminate. As far as I can tell it always has been, as we have never been able to predict which worker goroutine will process a task. If they were pulled off by different goroutines at roughly the same time, we don't know which one would "win" races to the locks etc.

A client making related requests should be waiting for responses anyway before proceeding onto the next request.